scss/bitstyles/tools/_font-size.scss

```diff
@@ -0,0 +1,42 @@
+@use 'sass:map';
+@use '../settings/typography';
+@use 'unit-conversions';
+
+// Palette
+//
+// **Returns requested font-size from the global `typography.$font-sizes` map.**
+// e.g. `font-size: font-size.get('h1');`
+//
+// @param {String} $font-size  The font-size, as named in the `typography.$font-sizes` map
+//
+// @param {String} $breakpoint   Optional. Defaults to `base`, you can request the font-size at other breakpoints as defined in the `typography.$font-sizes` map
+//
+
+@function get($font-size, $breakpoint: 'base') {
+  @if map.has-key(typography.$font-sizes, $breakpoint) {
+    $font-sizes: map.get(typography.$font-sizes, $breakpoint);
+    $base-size: typography.$font-size-base-small;
+
+    @if map.has-key($font-sizes, 'html') {
+      $base-size: map.get($font-sizes, 'html');
+    }
+
+    @if map.has-key($font-sizes, $font-size) {
+      @if $font-size == 'html' {
+        @return unit-conversions.px-to-rem(map.get($font-sizes, $font-size), 16px);
+      }
+
+      @else {
+        @return unit-conversions.px-to-rem(map.get($font-sizes, $font-size), $base-size);
```

Comment:
Is there a guarantee that the `font-size-base-small` is defined as `px`? Or what happens with this conversion if it's not in px?

Reply:
No, there’s no guarantee. And it’s not so easy to check unit type in sass (there’s a built-in function but it’s not guaranteed to output the same result in different platforms & sass implementations :/ )

We could require that the font-sizes are actually given in `rem` in the first place, wdyt?

Rems are pretty well-understood by devs by now, other libraries use them directly. It would mean we can remove this conversion totally. And it would remove the need for the other issue you raised here, about the 16px value - that’s only there for the conversion to rem 🤔

Reply:
I updated — fixed both problems you raised by specifying the font-sizes directly in rem 🧹

Reply:
Also an option. Wondering if that makes the upgrade path harder for existing projects though 🤔

Reply:
This isn’t a breaking change, existing projects could continue using `px` to specify font-sizes. Ideally they’d switch to using `rem` though, as that’s better CSS for the end user, so yes this adds a ‘thing to check’ to the list when updating. I think it’s not a big change tho, and the next release is a major version, so I’d go with it